src/merge_iterator.rs

```diff
@@ -38,6 +38,44 @@ impl<T1: KeyValueIterator, T2: KeyValueIterator> TwoMergeIterator<T1, T2> {
             self.iterator2.0.next_entry().await?,
         ))
     }
+
+    fn peek1(&self) -> Option<&RowEntry> {
+        self.iterator1.1.as_ref()
+    }
+
+    fn peek2(&self) -> Option<&RowEntry> {
+        self.iterator2.1.as_ref()
+    }
+
+    fn peek_inner(&self) -> Option<&RowEntry> {
+        match (self.peek1(), self.peek2()) {
+            (None, None) => None,
+            (Some(v1), None) => Some(v1),
+            (None, Some(v2)) => Some(v2),
+            (Some(v1), Some(v2)) => {
+                if v1.key < v2.key {
+                    Some(v1)
+                } else {
+                    Some(v2)
+                }
+            }
+        }
```

Comment:
I have an opened PR with a small helper for this exact case that should be merged soon:

https://github.com/slatedb/slatedb/pull/391/files#diff-23af5356f6001cd3993cd3c801fb7716ea02d1e504081b9fb569332db6107e80R142

same case for advance_iter below.

Reply:
i think some duplication is fine here? when the logic is obvious and code is not long,  readers need not to learn additional contexts from elsewhere 👀

Reply:
I feel this is the case iff it doesn't hurt readability. For me personally it's far more complex to interpret multiple match branches are "just merging options" if both of them are available, than seeing an explicitly named helper.

```
merge_options(self.peek1(), self.peek2(), |v1, v2| {
                if v1.key < v2.key {
                    v1
                } else {
                    v2
                }
})
```

Anyway, both of these are our personal preferences, so it would be great to align on a common approach throughout the codebase, since it seems to be quite frequently used pattern.

Reply:
i feel this `MergeTwoIterator` is more like an specialized version of the `MergeIterator`, with adhoc `if else` over `iter1` and `iter2` other than a more complex heap sort. imo we can even drop this `MergeTwoIterator` by always using the `MergeIterator`.

the intention of this logic here is to make a adhoc sort between two iters instead of merging two options. if we want extract common code from here, i'd hope to emphasis it's doing a `sort` instead of a `merge` here.

Reply:
> imo we can even drop this MergeTwoIterator by always using the MergeIterator

This would be nice. I wrote it this way because I wanted to be able to merge multiple iterators with different types, and I didn't know about `async-trait` at the time.